content/en/blogs/faas-openfunction.md

```diff
@@ -0,0 +1,300 @@
+---
+title: 'OpenFunction: Build a Novel Platform for Open-Source Function Computing from Scratch'  
+tag: 'Serverless, FaaS, OpenFunction'  
+keywords: Serverless, FaaS, OpenFunction  
+description: This article describes function computing concepts, reference architectures, and the latest trends in the cloud-native Serverless field, and further explains how OpenFunction builds a novel platform for open-source function computing from scratch.   
+createTime: '2021-12-03'  
+author: 'Huo Bingjie, Felix'  
+snapshot: 'https://pek3b.qingstor.com/kubesphere-community/images/serverless-faas-openfunction.png'
+---
+
+**Serverless computing**, commonly known as Serverless, has become a buzzword in the cloud-native field. It will fuel the next wave of development in cloud computing after IaaS and PaaS. Serverless emphasizes an architecture philosophy and service model that allows developers to focus on implementing business logics in applications rather than on managing infrastructures (e.g., servers). In its paper *Cloud Programming Simplified: A Berkeley View on Serverless Computing*, the University of California at Berkeley presents two key viewpoints on Serverless:
+
++ Although server-based computing will be here to stay, its importance will decrease as Serverless computing continues to mature.
++ Serverless computing will eventually become the paradigm of the cloud computing age, replacing the server-based computing model to a great extent and putting an end to the Client-Server age.
+
+Now, what is Serverless?
+
+## Introduction to Serverless
+
+On Serverless, the University of California at Berkeley also gave a clear definition in the above-mentioned paper: `Serverless computing = FaaS + BaaS`. Conventionally, cloud services are classified from bottom to top layer as Hardware, Base Cloud Platform, PaaS, and Applications according to the degree of abstraction. Nevertheless, the ideal state of the PaaS layer is to have the capability of Serverless. Thus, let's replace the PaaS layer with Serverless, which is indicated by the yellow part in the following figure.
+
+![](https://pek3b.qingstor.com/kubesphere-community/images/202112011620107.png)
+
+Serverless consists of two components, **BaaS** and **FaaS**. Base cloud services, including object storages, relational databases, and message queue (MQ), are essential for every cloud provider and belong to BaaS (Backend-as-a-Service). FaaS (Function as a Service) is the core of Serverless.
+
+## Analysis of Existing Open-Source Serverless Platforms
+
+The KubeSphere community has been conducting in-depth researches on Serverless since late 2020. We found that:
```

Comment:
Is `investigation` much suitable than `researches` here?

Reply:
May I ask why?

Reply:
research is also ok here in my opinion

Reply:
Np